[CI] CrossClusterEsqlRCS1EnrichUnavailableRemotesIT testEsqlEnrichWithSkipUnavailable failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #7695 / windows-2019_checkpart4_platform-support-windows](https://gradle-enterprise.elastic.co/s/3ywfnlsftkyi6)
- [elasticsearch-periodic-platform-support #7607 / rhel-7_platform-support-unix](https://gradle-enterprise.elastic.co/s/secsnvkbm6qdi)

**Reproduction Line:**
```
gradlew ":x-pack:plugin:security:qa:multi-cluster:javaRestTest" --tests "org.elasticsearch.xpack.remotecluster.CrossClusterEsqlRCS1EnrichUnavailableRemotesIT.testEsqlEnrichWithSkipUnavailable" -Dtests.seed=62C13C876F0CE32A -Dtests.locale=uz-Arab-AF -Dtests.timezone=Africa/Lusaka -Druntime.java=24
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.remotecluster.CrossClusterEsqlRCS1EnrichUnavailableRemotesIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testEsqlEnrichWithSkipUnavailable),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: test: esqlEnrichWithRandomSkipUnavailable: id
Expected: is <1>
     but: was <4>
```

**Issue Reasons:**
- [8.x] 2 failures in test testEsqlEnrichWithSkipUnavailable (0.4% fail rate in 471 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (14.3% fail rate in 14 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CrossClusterEsqlRCS1EnrichUnavailableRemotesIT.java
+++ b/x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/CrossClusterEsqlRCS1EnrichUnavailableRemotesIT.java
@@ -30,6 +30,7 @@ import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.endsWith;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.hasItemInArray;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.oneOf;
 
@@ -73,6 +74,9 @@ public class CrossClusterEsqlRCS1EnrichUnavailableRemotesIT extends AbstractRemo
     public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
 
     private final String[] modes = { "_coordinator", "_remote" };
+
+    // These are the IDs used in the test data in setSourceData().
+    private final Integer[] employeeIDs = { 1, 2, 3, 4, 5, 6 };
 
     @Before
     public void setupPreRequisites() throws IOException {
@@ -121,7 +125,7 @@ public class CrossClusterEsqlRCS1EnrichUnavailableRemotesIT extends AbstractRemo
             // Email
             assertThat(info.apply("email was: " + value.get(0)), (String) value.get(0), endsWith("@corp.co"));
             // ID
-            assertThat(info.apply("id"), value.get(1), is(i + 1));
+            assertThat(info.apply("id"), employeeIDs, hasItemInArray((int) value.get(1)));
         }
 
         assertThat(info.apply("total clusters"), (int) clusters.get("total"), is(2));
@@ -170,7 +174,7 @@ public class CrossClusterEsqlRCS1EnrichUnavailableRemotesIT extends AbstractRemo
                 // Email
                 assertThat(info.apply("email was: " + value.get(0)), (String) value.get(0), endsWith("@corp.co"));
                 // ID
-                assertThat(info.apply("id"), value.get(1), is(i + 1));
+                assertThat(info.apply("id"), employeeIDs, hasItemInArray((int) value.get(1)));
             }
 
             assertThat(info.apply("total clusters"), (int) clusters.get("total"), is(2));

